### ISSUE TITLE
No result from the test

### ISSUE THREAD

**Issue by baiyu001** [hellloxiaotian/HDSRNet#1]
Hello, thank you very much for open sourcing such great work, but I have a few more questions to ask:
![1709629784812(1)](https://github.com/hellloxiaotian/HDSRNet/assets/82936454/d0bca86b-657f-4333-89d0-2a6f5c9bf15e)
    Is the 'sample.py' file mentioned here not published yet and I can't find it? And I had some problems testing my data set using the pre-trained weights provided. I suspect that the image input path and image output save path I set differ from the source code. Can you give a more specific example?
    Looking forward to your reply!

**Comment by hellloxiaotian** [comment]
Thank you for your attention!
1. We have modified ‘sample.py’ to ‘option.py’.
2. We  have shown the path of our training data in the updated 'HDSRNet/data/div2k.py', this may help you. In this work, we used 900 images in DIV2K both traing and validation dataset, you can see more relevant contents in Section IV.A Training datasets of the paper :
'A Heterogeneous Dynamic Convolutional Neural Network for Image Super-resolution', 'https://arxiv.org/abs/2402.15704'.

**Comment by baiyu001** [comment]
> Thank you for your attention!
> 
> 1. We have modified ‘sample.py’ to ‘option.py’.
> 2. We  have shown the path of our training data in the updated 'HDSRNet/data/div2k.py', this may help you. In this work, we used 900 images in DIV2K both traing and validation dataset, you can see more relevant contents in Section IV.A Training datasets of the paper :
>    'A Heterogeneous Dynamic Convolutional Neural Network for Image Super-resolution', 'https://arxiv.org/abs/2402.15704'.

**Comment by baiyu001** [comment]
> Thank you for your attention!
> 
> 1. We have modified ‘sample.py’ to ‘option.py’.
> 2. We  have shown the path of our training data in the updated 'HDSRNet/data/div2k.py', this may help you. In this work, we used 900 images in DIV2K both traing and validation dataset, you can see more relevant contents in Section IV.A Training datasets of the paper :
>    'A Heterogeneous Dynamic Convolutional Neural Network for Image Super-resolution', 'https://arxiv.org/abs/2402.15704'.

OK, thank you very much for your reply ！

**Comment by baiyu001** [comment]
Hello, sorry to bother you again, I found something wrong when debugging the code, I think it may be caused by some reasons, so I would like to bring it up and discuss with you:
![dend_background](https://github.com/hellloxiaotian/HDSRNet/assets/82936454/ba21d3e5-2e4a-4a61-b905-62f722078f9e)
![end_background](https://github.com/hellloxiaotian/HDSRNet/assets/82936454/9142bd45-9e81-45de-b0f5-0279c8caf1e9)
Figure 1 shows the code defined in the ./HDSRNet/utility.py -> class checkpoint() -> dend_background() function. The function defined in line 146 of the figure is dend_background(). 
Figure 2 shows the code called in the ./HDSRNet/trainer.py -> class Trainer() -> test() function. The end_background() on line 142 does not seem to match the dend_background() name on line 146 in Figure 1, which may cause an error in the code. I suggest changing the function definition name or changing it to the code on line 143 in Figure 2.
The above is my personal opinion, if there is not correct, please criticize and correct, looking forward to your reply!

**Comment by baiyu001** [comment]
After your guidance, I have read all the contents in IV.A of the paper in detail, and I have configured the storage path of the data set according to the parameters in 'option.py'. However, when I used the test command in 'README.md' to test the code, I found that the output image of the model could not be obtained successfully. I don't know whether there is a problem with my data reading path or the data saving path of model output. Could you please tell me in detail what is the correct image input path and image output saving path?
Please tolerate my stupidity and look forward to your reply!

**Comment by hellloxiaotian** [comment]
Thank you for your question, we have updated the trainer.py.

The path of the training dataset:
The print from the 25th, 34th and 41th line in data/div2k.py are:
dir_hr: /data/zxy/datasets/DIV2K/DIV2K_train_HR
dir_lr: /data/zxy/datasets/DIV2K/DIV2K_train_LR_bicubic
names_lr ['/data/zxy/datasets/DIV2K/DIV2K_train_LR_bicubic/X4/0001x4.png', ......]

The path of the output Set5 SR results is: '/data/zxy/codes/DCSRN-TCE-github/experiment/test/results-Set5/'.

Hope this can solve your problem.

**Comment by baiyu001** [comment]
Thank you for your reply !
The path of the my dataset:
train_lr : "/data/HDSRNet-main/my_datasets/DIV2K/DIV2K_train_LR_bicubic/X4/img_001_SRF_4_LR.png, ......"
train_hr : "/data/HDSRNet-main/my_datasets/DIV2K/DIV2K_train_HR/X4/img_001_SRF_4_HR.png, ......"
test_lr : "/data/HDSRNet-main/my_datasets/Set5/LRbicx4/img_001_SRF_4_LR.png"
test_hr : "/data/HDSRNet-main/my_datasets/Set5/original/bin/img_001_SRF_4_HR.png"
The option.py is configured as:
'--dir_data'='/data/HDSRNet-main/my_datasets', '--data_train'='DIV2K', '--data_test'='Set5', '--save'='test', '--load'=' '
I run the code using the instructions in the README.md:
python main.py --model hdsrnet --scale 4  --data_test Set5  --pre_train ../checkpoint/HDSRNet_x4.pt --test_only --save_results
But I can't get the SR output result of the model in the path of "/data/HDSRNet-main/experiment/test/results-Set5/", could you please help me correct the problem?
I am very sorry to bother you again and look forward to your reply！

**Comment by hellloxiaotian** [comment]
I can't see any problem in your description. 
Do you meet any error information in your test processing? You can show them with screenshot like before.
Did a progress bar based on the number of test set images appear normally during the testing process? For example, '1/5, 2/5, ... 5/5' with Set5 dataset.

Everything goes well on our computer.
![github](https://github.com/hellloxiaotian/HDSRNet/assets/25679314/b2ebe2dc-8cc0-4d82-aeaa-d4e5a6677c86)
![image](https://github.com/hellloxiaotian/HDSRNet/assets/25679314/3315b329-00a6-4eb3-9f5a-eccd3cd90969)

**Comment by baiyu001** [comment]
I did not encounter any error message during the test, but the progress bar based on the number of images in the test set did not appear normally during the test, and I could not successfully get the output image on my computer, I do not know what the problem is
![test_result](https://github.com/hellloxiaotian/HDSRNet/assets/82936454/a79adc78-036f-4cc5-953b-5dbb8c5e98e1)

**Comment by hellloxiaotian** [comment]
The progress bar is missed usually means your image path was wrong. The program did not read the test images.

**Comment by baiyu001** [comment]
Ok, thank you for your careful answer, I will check again in the future.

**Comment by haojin0323** [comment]
Hello, I have the same question. My running test cannot get the results in the directory "/data/HDSRNet-main/experiment/test/results-Set5".(This step generates a vacant directory) The following is my screenshot.
![image](https://github.com/hellloxiaotian/HDSRNet/assets/73220321/0c023240-83fc-475d-8422-70ec2e0cdc3b)
The path of my dataset:
/data/zxy/datasets/Set5/LRbicx4/img_001_SRF_4_LR.png,...
/data/zxy/datasets/Set5/original/bin/img_001_SRF_4_HR.png,...
The option.py is configured as:
'--dir_data':'/data/zxy/datasets/'
The command I use is as follows:
python main.py --model hdsrnet --scale 4 --data_test Set5 --pre_train ../checkpoint/HDSRNet_x4.pt --test

**Comment by XuanyuZhang** [comment]
Thank you for your attention!

In your screenshot, I wonder if there is a problem in your path? The 'data' may be replaced by 'DATA' ?
![image](https://github.com/hellloxiaotian/HDSRNet/assets/35928688/6f4273ec-39b8-4714-99db-b79f5fd61f84)

The evaluation:0 usually means your program do not load the relative data.
![image](https://github.com/hellloxiaotian/HDSRNet/assets/35928688/a12149e3-9b1f-4972-a5f2-4ebe7d8c454a)

**Comment by 2029686068** [comment]
> Hello, I have the same question. My running test cannot get the results in the directory "/data/HDSRNet-main/experiment/test/results-Set5".(This step generates a vacant directory) The following is my screenshot. ![image](https://private-user-images.githubusercontent.com/73220321/320259332-0c023240-83fc-475d-8422-70ec2e0cdc3b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjMzNjU4MDAsIm5iZiI6MTcyMzM2NTUwMCwicGF0aCI6Ii83MzIyMDMyMS8zMjAyNTkzMzItMGMwMjMyNDAtODNmYy00NzVkLTg0MjItNzBlYzJlMGNkYzNiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODExVDA4MzgyMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY5Mjk0Y2E3ODhhZjRkYmZhODE4NjE0M2RmNTA2MTg1ZTE0MTg1NDg1YTZhYWNkNDM1ZjAwOTM3MTBkMGYzMDImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.xSc2kw392ZZ_jafKGsE3wHJ7qolLxDbqw_PqqtezJU4) The path of my dataset: /data/zxy/datasets/Set5/LRbicx4/img_001_SRF_4_LR.png,... /data/zxy/datasets/Set5/original/bin/img_001_SRF_4_HR.png,... The option.py is configured as: '--dir_data':'/data/zxy/datasets/' The command I use is as follows: python main.py --model hdsrnet --scale 4 --data_test Set5 --pre_train ../checkpoint/HDSRNet_x4.pt --test

I also encountered this problem, how did you solve it
